spanner/session.go

```diff
@@ -1364,7 +1450,28 @@ func (hc *healthChecker) maintainer() {
 			hc.pool.recordStat(context.Background(), MaxInUseSessionsCount, int64(hc.pool.maxNumInUse))
 			hc.pool.lastResetTime = now
 		}
+
+		// task that fetches sessions which are unexpectedly long-running
+		if now.After(hc.pool.InactiveTransactionRemovalOptions.lastExecutionTime.Add(hc.pool.executionFrequency)) {
+			if hc.pool.CloseInactiveTransactions || hc.pool.LogInactiveTransactions {
+				longRunningSessions = hc.pool.getLongRunningSessionsLocked()
+			}
+			hc.pool.InactiveTransactionRemovalOptions.lastExecutionTime = now
+		}
 		hc.pool.mu.Unlock()
+
+		// destroy long-running sessions
+		if hc.pool.CloseInactiveTransactions {
+			for _, sh := range longRunningSessions {
+				// removes inner session out of the pool to reduce the probability of two processes trying
+				// to use the same session at the same time.
+				sh.destroy()
```

Comment:
Just to be sure: Does this `destroy` method take care of all house-keeping regarding the number of sessions in the pool etc?